### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by 15lnestor**
## Description of Change
Adds controllers to the application.

## Alternate Designs
N/A

## Benefits
N/A

## Possible Drawbacks
N/A

### CODE REVIEW

File: app/controllers/leagues_controller.rb [hunk 1]
```diff
@@ -21,8 +22,20 @@ def show
...
+  def edit
+    @league = League.find(params[:id])
+  end
+
   private
     def league_params
       params.require(:league).permit(:name)
     end
+
+    def require_permission
```
Comment (scottmascio2115) [inline app/controllers/leagues_controller.rb]: Just taking a quick glance. This isnt my full pr review.

But we use `Devise` at cmm for authentication and `CanCanCan` for user permissions check it out https://github.com/CanCanCommunity/cancancan it maybe useful. Definitely worth learning about at some point

File: spec/controllers/leagues_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      end
+    end
+  end
+
+  describe 'PATCH #update' do
+    context 'when you own the league' do
+      it 'updates the league' do
+        sign_in user
+        patch :update, params: { id: user_league.id, league: { name: "Updated Name" } }
+        # expect(user_league.name).to eq("Updated Name") Not working but the name definitely changes
```
Comment (15lnestor) [inline spec/controllers/leagues_controller_spec.rb]: There are several cases where I've encountered problems changing something defined in a let block. In this case, I have manually tested that the name changed, but the commented assertion fails. I suspect i may have to do with the way the let block works.

Comment (scottmascio2115) [reply spec/controllers/leagues_controller_spec.rb]: Try `user_league.reload.name`

**Comment by 15lnestor** [comment]
OK, I think I'm ready, so whenever you have time. I only did the leagues controller here because I can foresee a lot of changes needing to be made, mostly with the tests. Once we get this one flushed out, the other controllers should go smoothly.

File: spec/controllers/leagues_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    context 'when not signed in' do
+      it 'renders the index template' do
+        get :index
+        expect(response).to be_success
+      end
+    end
+
+    context 'when signed in' do
+      it 'renders the index template' do
+        sign_in user
```
Comment (scottmascio2115) [inline spec/controllers/leagues_controller_spec.rb]: Its my preference to put any set up in a before block. Just reads cleaner to me.

```
context 'when signed in' do
  before { sign_in user }
  it 'renders the index template' do
    get :index
    expect(response).to be_success
```

File: spec/controllers/leagues_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,153 @@
+require 'rails_helper'
+
+RSpec.describe LeaguesController, type: :controller do
```
Comment (scottmascio2115) [inline spec/controllers/leagues_controller_spec.rb]: Any of the permission edge cases to test besides just signed in not signed in?

File: spec/controllers/leagues_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    context 'when you own the league' do
+      it 'deletes the league' do
+        sign_in user
+        user_league # Need to load this before the next call
```
Comment (scottmascio2115) [inline spec/controllers/leagues_controller_spec.rb]: You could do `let!`

File: app/views/leagues/edit.html [hunk 6]
```diff
@@ -0,0 +1 @@
+<h1>Edit League</h1>
```
[the remark this answers is not in the document]
Comment (scottmascio2115) [reply app/views/leagues/edit.html]: Same with twitter bootstap?

[the remark this answers is not in the document]
Comment (15lnestor) [reply app/views/leagues/edit.html]: Yeah I'm going to add those. I was mostly interested in the controllers here, not the views. Also haml adds some weird deprecation warning, so I'm going to look into how to get rid of that.
